ichnaea/db.py

```diff
@@ -17,36 +16,47 @@
 from ichnaea.conf import settings
 
 
-DB_TYPE = {
-    "ddl": settings("db_ddl_uri"),
-    "ro": settings("db_readonly_uri"),
-    "rw": settings("db_readwrite_uri"),
-}
+DB_TYPE = {"ro": settings("db_readonly_uri"), "rw": settings("db_readwrite_uri")}
 
 
-def get_alembic_config():
-    cfg = AlembicConfig()
+class SqlAlchemyUrlNotSpecified(Exception):
+    """Raised when SQLALCHEMY_URL is not specified in environment."""
 
-    script_location = os.path.join(os.path.dirname(__file__), "alembic")
-    assert os.path.exists(script_location)
-    cfg.set_section_option("alembic", "script_location", script_location)
-    cfg.set_section_option("alembic", "sqlalchemy.url", settings("db_ddl_uri"))
-    return cfg
+    def __init__(self, *args, **kwargs):
+        super().__init__("SQLALCHEMY_URL is not specified in the environment")
 
 
-def configure_db(type_=None, uri=None, _db=None):
+def get_sqlalchemy_url():
+    """Returns the ``SQLALCHEMY_URL`` environment value.
+
+    :returns: the sqlalchemy url to be used for alembic migrations
+
+    :raises SqlAlchemyUrlNotSpecified: if ``SQLALCHEMY_URL`` was not specified
+        or is an empty string
+
+    """
+
+    url = os.environ.get("SQLALCHEMY_URL", "")
+    if not url:
+        raise SqlAlchemyUrlNotSpecified()
+    return url
+
+
+def configure_db(type_=None, uri=None, _db=None, pool=True):
     """
     Configure and return a :class:`~ichnaea.db.Database` instance.
 
+    :param type_: one of "ro" or "rw"
+    :param uri: the uri to connect to; if not provided, uses ``type_``
+        parameter
     :param _db: Test-only hook to provide a pre-configured db.
+    :param pool: True for connection pool, False for no connection pool
+
```

Comment:
At this point, this is kind of a mess of a function. I thought about trying to make it less messy, but didn't have any good ideas that seemed worthwhile to spend time on. I figure we can push this off to later.